SC_redesign/Crop_and_Soil/soil/soil_temp.py

```diff
@@ -0,0 +1,544 @@
+#
+#
+# def calc_albedo(soil, crop_type):
+#     """
+#     Description:
+#         Calculates the daily albedo as a function of soil type, plant cover,
+#         and snow cover.
+#         "pseudocode_soil" S.1.A.9/10
+#
+#     Args:
+#         soil: an instance of the Soil class
+#         crop: an instance of the Crop class
+#
+#     Returns:
+#         int: soil albedo
+#     """
+#
+#     CV = crop_type.bio_AG
+#
+#     # "pseudocode_soil" S.1.A.10
+#     cover = exp(-0.00005 * CV)
+#
+#     # "pseudocode_soil" S.1.A.9
+#     return 0.23 * (1 - cover) + soil.soil_albedo * cover
+#
+#
+# def calc_bcv(crop_type, time):
+#     """
+#     Description:
+#         Calculates the weighting factor for ground cover
+#         "pseudocode_soil" S.1.A.11/12
+#
+#     Args:
+#         crop: an instance of the Crop class
+#         time: an instance of the Time class
+#
+#     Returns:
+#         int: bcv, the bio-cover weighting factor for ground cover
+#     """
+#
+#     CV = crop_type.bio_AG
+#     exp_part = exp(7.563 - 0.0001297 * (-CV))
+#
+#     bcv = CV / (CV + exp_part)
+#
+#     snow = 0
+#     # TODO: arbitrary snow flag - GitHub Issue #164
+#     if time.day > 335 or time.day < 59:
+#         albedo_snow = 0.8
+#         snow = 10 * albedo_snow
+#
+#     bcv_snow = (snow / (snow + exp(6.055 - 0.3002 * snow)))
+#
+#     return max(bcv, bcv_snow)
+from typing import Optional
+from math import exp, log
+
+from SC_redesign.Crop_and_Soil.soil.soil_data import SoilData
+
+"""
+This module is based on the "Soil Temperature" section of SWAT (1:1.3.3)
+"""
+class SoilTemp:
+
+    def __init__(self, soil_data: Optional[SoilData] = None):
+        self.data = soil_data or SoilData()
+
+    def daily_soil_temperature_update(self, solar_radiation: float, avg_temp: float, min_temp: float, max_temp: float,
+                                      plant_cover: float, snow_cover: float, avg_annual_air_temp: float) -> None:
+        """this is the main routine that updates the soil temperature
+
+        Args:
+            solar_radiation: solar radiation reaching the ground on a given day (MJ per square meter per day)
+            avg_temp: average temperature of a given day (degrees C)
+            min_temp: minimum temperature of a given day (degrees C)
+            max_temp: maximum temperature of a given day (degrees C)
+            plant_cover: total aboveground plant biomass and residue on a given day (kg per hectare)
+            snow_cover: water content of the snow cover on a given day (mm)
+            avg_annual_air_temp: average annual air temperature (degrees C)
+
+        Important: SWAT does not specify how to start the simulation i.e. it does not specify what to do on day 0, when
+            there is no previous day's temperature. Currently, the implementation just uses the temperature that the
+            soil starts (it sets the previous days temperature equal to the current day's temperature)
```

Comment:
Further elaborating:

```suggestion
        Important: SWAT does not specify how to start the simulation i.e. it does not specify what to do on day 0, when
            there is no previous day's temperature. Currently, the implementation just uses the temperature that the
            soil starts (it sets the previous days temperature equal to the current day's temperature). This assumption
            is fairly reasonable due to temporal auto-correlation, but does not account for the random fluctuations
            that can occur throughout the year.
```